### PR TITLE
8343004: Adjust JAXP limits

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Adjust JAXP Limits. Limits are adjusted as specified in the CSR.

Tests: 
     Updated the config test with the new settings.

     Removed obsolete tests Bug6309988.java and Bug4674384_MAX_OCCURS_Test.java, and files used by these tests. These tests verified legacy properties. They came into the JDK from the JAXP and SJSXP standalone libraries. Improvements were made before the integration into the JDK via for example, SJSXP Issue 30. Though the issue tracker is no longer accessible, the tests, e.g. IssueTracker30, are in the JAXP test suite.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8343006](https://bugs.openjdk.org/browse/JDK-8343006) to be approved

### Issues
 * [JDK-8343004](https://bugs.openjdk.org/browse/JDK-8343004): Adjust JAXP limits (**Enhancement** - P4)
 * [JDK-8343006](https://bugs.openjdk.org/browse/JDK-8343006): Adjust JAXP limits (**CSR**)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22337/head:pull/22337` \
`$ git checkout pull/22337`

Update a local copy of the PR: \
`$ git checkout pull/22337` \
`$ git pull https://git.openjdk.org/jdk.git pull/22337/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22337`

View PR using the GUI difftool: \
`$ git pr show -t 22337`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22337.diff">https://git.openjdk.org/jdk/pull/22337.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22337#issuecomment-2495082799)
</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]
Hi Joe,

The changes look good overall

**Review by RogerRiggs — APPROVED** [review]
Looks fine

File: src/java.xml/share/conf/jaxp-strict.properties.template [hunk 1]
```diff
@@ -96,15 +96,15 @@ jdk.xml.maxParameterEntitySizeLimit=15000
...
 # Limits the number of attributes an element can have. The default value is 10000.
-jdk.xml.elementAttributeLimit=10000
+jdk.xml.elementAttributeLimit=200
 #
 # Limits the number of content model nodes that may be created when building a
 # grammar for a W3C XML Schema that contains maxOccurs attributes with values
 # other than "unbounded". The default value is 5000.
 jdk.xml.maxOccurLimit=5000
 #
 # Limits the maximum element depth. The default value is 0.
```
Comment (seanjmullan) [inline src/java.xml/share/conf/jaxp-strict.properties.template:106]: s/0/100/

File: src/java.xml/share/conf/jaxp-strict.properties.template [hunk 2]
```diff
@@ -96,15 +96,15 @@ jdk.xml.maxParameterEntitySizeLimit=15000
 jdk.xml.entityReplacementLimit=100000
 #
 # Limits the number of attributes an element can have. The default value is 10000.
```
Comment (seanjmullan) [inline src/java.xml/share/conf/jaxp-strict.properties.template:98]: s/10000/200/

File: test/jaxp/javax/xml/jaxp/unittest/common/config/ImplProperties.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ */
+package common.config;
+
+/**
+ * Implementation Specific Properties:
+ * https://docs.oracle.com/en/java/javase/23/docs/api/java.xml/module-summary.html#IN_ISFP
+ */
+public class ImplProperties {
+    // JDK Implementation Specific Properties, refer to module description
+    public static final String[] PROPERTY_KEYS = {
```
[the remark this answers is not in the document]
Comment (JoeWang-Java) [reply test/jaxp/javax/xml/jaxp/unittest/common/config/ImplProperties.java:31]: Will look into that. Added to JDK-8344999.

**Review by LanceAndersen — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Pull request closed without merging**